### PR TITLE
Plugins: Ignore trailing slash in root URL check

### PR DESCRIPTION
**Pull request by wbrowne**
This fixes an issue where the manifest root URL and the application URL can differ in the path/URI by a trailing slash.

For example:

```
"rootUrls": [
    "http://localhost:3000/"
],
```

and 
```
cfg.AppURL = http://localhost:3000
```

Or vice versa. The change will use `path.Clean` to ensure a more sane check.

This also changes the root URL check to incorporate the `AppSubUrl` config that is used as a optional prefix to all URLs

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM -- but it seems we should force a trailing slash in the original service also

**Comment by wbrowne** [comment]
@ryantxu do you mean the toolkit should always ensure a trailing slash is generated for the root URL?

File: pkg/plugins/manager/manager_test.go [hunk 1]
```diff
@@ -282,6 +282,32 @@ func TestPluginManager_Init(t *testing.T) {
 		assert.Nil(t, pm.plugins[("test")])
 	})
 
+	t.Run("With back-end plugin with valid v2 private signature (plugin root URL ignores trailing slash)", func(t *testing.T) {
+		origAppURL := setting.AppUrl
+		t.Cleanup(func() {
+			setting.AppUrl = origAppURL
+		})
+		setting.AppUrl = "http://localhost:3000/grafana/"
```
Comment (marefr) [inline pkg/plugins/manager/manager_test.go:290]: This should perhaps be split up in `setting.AppUrl` and `setting.AppSubUrl` to be more close to reality? Based on https://github.com/grafana/grafana/blob/1f5a28ec63c3a6d0748aa58c08f4b25e166c0ccd/pkg/setting/setting.go#L1364-L1371

**Review by marefr — APPROVED** [review]
LGTM, but added a comment

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**